### ISSUE TITLE
AM2PG crashes with LWJGL3ify and Java 21.

### ISSUE THREAD

**Issue by lordmuff** [AirBurn0/am2playground#17]
Description. Try running AM2PG with the latest version of LWJGL3ify and all the dependencies, and you will see a crash. Seems to be some method you're using that was removed from newer versions of Java. Would love to see something done about this as it's the only mod in my modpack which is straight up incompatible with LWJGL3ify and Java 21. Will supply more detail if necessary.

**Comment by AirBurn0** [comment]
> Description. Try running AM2PG with the latest version of LWJGL3ify and all the dependencies, and you will see a crash.

Well, it is.
I already made a huge refactoring including replacing some reflection API in favor of Access Transformers (because I can), so I don’t even need to fix anything.
However, in order for the new version to work, you need to add the following lines to the array in the **'config/lwjgl3ify-early.json'** file:
```
    "am2.api.spell.enums.SpellModifiers",
    "am2.api.spell.enums.ContingencyTypes"
```
![image](https://github.com/user-attachments/assets/3b02af96-f71a-4510-a37a-f18835824880)

**Comment by lordmuff** [comment]
Thank you. I will try this.

**Comment by lordmuff** [comment]
Also, is it intentional that this repository doesn't seem to include any actual CODE? For me all that I see is localization stuff.

**Comment by AirBurn0** [comment]
> Also, is it intentional that this repository doesn't seem to include any actual CODE? For me all that I see is localization stuff.

https://github.com/AirBurn0/am2playground/blob/0f97474238f1cfa97fa77b6d2878f88aac165087/README.md?plain=1#L55C1-L59C1

## TLDR:
Project is NOT opensource because there are jokes in the code that may offend someone (really only reason).

**Comment by lordmuff** [comment]
Odd decision IMO, since there's no GitHub beholding you to remove anything offensive as long as it isn't literal malware. AFAIK really rude code comments are kinda par for the course with programming.

Regardless, following your instructions led to my instance successfully booting, but I had a crash when loading into the world which I figure is unrelated to this specific issue. However, I'll include the logs here for posterity. Otherwise this issue is probably considered closed on my end.
[fml-client-latest.log](https://github.com/user-attachments
[crash-2024-07-13_16.54.43-server.txt](https://github.com/user-attachments/files/16203650/crash-2024-07-13_16.54.43-server.txt)
[crash-2024-07-13_16.54.48-client.txt](https://github.com/user-attachments/files/16203651/crash-2024-07-13_16.54.48-client.txt)
/files/16203649/fml-client-latest.log)

**Comment by AirBurn0** [comment]
> Odd decision IMO, since there's no GitHub beholding you to remove anything offensive as long as it isn't literal malware. AFAIK really rude code comments are kinda par for the course with programming.
> 
> Regardless, following your instructions led to my instance successfully booting, but I had a crash when loading into the world which I figure is unrelated to this specific issue. However, I'll include the logs here for posterity. Otherwise this issue is probably considered closed on my end.
> 
> [crash-2024-07-13_16.54.43-server.txt](https://github.com/user-attachments/files/16203650/crash-2024-07-13_16.54.43-server.txt)
> 
> [crash-2024-07-13_16.54.48-client.txt](https://github.com/user-attachments/files/16203651/crash-2024-07-13_16.54.48-client.txt)
> 
> [fml-client-latest.log](https://github.com/user-attachments/files/16203649/fml-client-latest.log)

Well, I'll try to look at this, because beta version isn't fully compatable with AM2.5 (I'm developing on original Mithion AM2  v1.4.009 to ensure maximum compatibility), but release will be.

**Comment by lordmuff** [comment]
I respect your decision, but you could always just... yknow. Remove the comments. If you did something REALLY racist. I dunno, I guess I just don't get it.

Also, be warned: the modpack I tested with this is quite the edge case, seeing as I'm running GregTech 6, 80% of GT:NH, Reika's mods, and HBM's Nuclear Tech... **all at the same time.** So it may not be representative of an actual error. Your Mileage May Vary.

**Comment by AirBurn0** [comment]
> I respect your decision, but you could always just... yknow. Remove the comments.

Yeah. The point is - I don't want. Not now, at least. This will require refactoring, which is what I'm doing... sometimes. 
You know, you can either **write new** code or **rewrite the old** one. And I don’t think that most of my addon users care about my code, but do care about new content that I'm baking rn.

> If you did something REALLY racist. I dunno, I guess I just don't get it.

...war related. I'm Russian, after all.

> Also, be warned: the modpack I tested with this is quite the edge case, seeing as I'm running GregTech 6, 80% of GT:NH, Reika's mods, and HBM's Nuclear Tech... **all at the same time.** So it may not be representative of an actual error. Your Mileage May Vary.

Well, yes, but... Reika would never bother to fix this, especially because not only Reika mod was the case. GTNH dudes may, but... Since AM2PG isn't part of GTNH modpack, likely they won't either. For other mods - their authors are most likely either dead (I mean, they haven’t been in touch for a long time), or they don’t really care anymore.

So who should fix this if not me? Show me this unfortunate guy ASAP - I have a lot of work that I want to dump on someone.

**Comment by lordmuff** [comment]
Y'know what, all of that is completely fair. I concede. Thanks for entertaining my questions, lmao. I eagerly await whatever you might do in the future. Apologies for any unintended rudeness.
